terminus-ui/chip/src/chip-collection.component.ts

```diff
@@ -597,44 +598,40 @@ export class TsChipCollectionComponent implements AfterViewInit, OnInit, OnDestr
     });
   }
 
+
   /**
    * Listens to remove events on each chip.
    */
-
   private listenToChipsRemoved(): void {
-    this.chipRemoveSubscription = this.chipRemoveChanges.subscribe(event => {
+    this.chipRemoveSubscription = this.chipDestroyChanges.subscribe(event => {
       const chip = event.chip;
       const chipIndex = this.chips.toArray().indexOf(event.chip);
 
-      // In case the chip that will be removed is currently focused, we temporarily store
-      // the index in order to be able to determine an appropriate sibling chip that will
-      // receive focus.
+      // In case the chip that will be removed is currently focused, we temporarily store the index in order to be able to determine an
+      // appropriate sibling chip that will receive focus.
+      // istanbul ignore else
       if (this.isValidIndex(chipIndex) && chip.hasFocus) {
         this.lastDestroyedChipIndex = chipIndex;
       }
+      this.removed.emit(new TsChipEvent(chip));
     });
   }
 
+
   /**
    * Syncs the collection's state with the individual chips.
    */
-  private syncChipsState() {
-    if (this.chips) {
+  private syncChipsState(): void {
+    // istanbul ignore else
+    if (this.chips && this.chips.length) {
       this.chips.forEach(chip => {
-        chip.isDisabled = this.isDisabled;
+        chip.allowMultiple = this.allowMultipleSelections;
         chip.chipCollectionMultiple = this.allowMultipleSelections;
+        chip.isDisabled = this.isDisabled;
         chip.isRemovable = !this.isReadonly && !this.isDisabled;
+        chip.isSelectable = this.isSelectable;
```

Comment:
Moved this here rather than the setter since it wasn't getting reflected to the chips if the flag was set before the chips where collected.

Reply:
👍